src/components/InternalTransactionFlatTable.vue

```diff
@@ -134,24 +135,43 @@ export default {
                 this.updateData();
             },
         },
+        // each time the address changes, we clear the pagination
+        address() {
+            this.clearPagination();
+        },
     },
     methods: {
+        clearPagination() {
+            this.pagination = {
+                ...this.pagination,
+                page: 1,
+                rowsPerPage: this.initialPageSize,
+                rowsNumber: 0,
+            };
+        },
         updateData() {
-            const _pag = this.$route.query.page;
-            let pag = _pag;
-            let page = 1;
-            let size = this.page_size_options[0];
+            clearTimeout(this.timer);
+            this.timer = setTimeout(() => {
+                // we need to wait a bit to allow the URL to be updated
+                const _pag = this.$route.query.page;
+                let pag = _pag;
+                let page = 1;
+                let size = this.page_size_options[0];
 
-            // we also allow to pass a single number as the page number
-            if (typeof pag === 'number') {
-                page = pag;
-            } else if (typeof pag === 'string') {
-                // we also allow to pass a string of two numbers: 'page,rowsPerPage'
-                const [p, s] = pag.split(',');
-                page = p;
-                size = s;
-            }
-            this.setPagination(page, size);
+                // we also allow to pass a single number as the page number
+                if (typeof pag === 'number') {
+                    page = pag;
+                } else if (typeof pag === 'string') {
+                    // we also allow to pass a string of two numbers: 'page,rowsPerPage'
+                    const [p, s] = pag.split(',');
+                    page = p;
+                    size = s;
+                } else {
+                    // we are in the first page loading for the first time, so we clear the pagination
+                    this.clearPagination();
+                }
```

Comment:
No pagination parameter in the URL -> clearPagination() - the final part of the fix